File: PROG1 and PROG2/08_Praktikum_Hochschule-master/src/Person.java
public class Person {
    private String name;
    private String id;

    protected Person() {
        name = null;
        id = null;
    }
    protected Person(String name, String id) {
        this.name = name;
        this.id = id;
    }

    public String gibInfo(){
        return name + ", ID" + id;
    }
}

File: snp-lab-code/P06_Personen_Verwaltung_Linked_List/work/personen-verwaltung/src/person.c
#include <assert.h>
#include <string.h>
#include <stdio.h>

#include "person.h"

int person_compare(const person_t *a, const person_t *b) {
    assert(a);
    assert(b);
     int result = strcmp(a->name, b->name);
     if (result == 0) {
        result = strcmp(a->first_name, b->first_name);
    }
    if (result == 0) {
        result = a->age - b->age;
    }
    return result;
}

int person_read(person_t *p) {
    assert(p);
    assert(NAME_LEN == 20);
    return scanf("%19s %19s %u", p->name, p->first_name, &(p->age)) == 3;
}
